packages/nouns-webapp/src/hooks/useBuyNoun.ts

```diff
@@ -0,0 +1,62 @@
+import { useContractFunction, useEthers } from '@usedapp/core';
+import { useAppDispatch, useAppSelector } from '../hooks';
+import { setAlertModal } from '../state/slices/application';
+import { Contract } from '@ethersproject/contracts';
+import { useEffect, useMemo, useState } from 'react';
+import config from '../config';
+import { getVrgdaAuctionContract } from '../utils/vrgdaAuction';
+
+export function useBuyNoun() {
+  const [nounId, setNounId] = useState(0);
+  const activeAuction = useAppSelector(state => state.auction.activeAuction);
+  const activeAccount = useAppSelector(state => state.account.activeAccount);
+
+  const { library } = useEthers();
+
+  const dispatch = useAppDispatch();
+
+  const auctionContract: Contract = useMemo(() => {
+    return getVrgdaAuctionContract(library);
+  }, [library]);
+
+  const { send, state } = useContractFunction(auctionContract, 'buyToken');
+
+  async function buyToken(blockNumber: number) {
+    try {
+      if (!activeAccount) throw new Error(`Please login`);
+
+      if (!activeAuction) throw new Error(`Couldn't get data about active auction`);
+      const { amount, nounId } = activeAuction;
+
+      setNounId(nounId.toNumber());
+
+      console.debug('buyToken call', { blockNumber, amount, nounId });
+
+      await send({
+        payableAmount: amount,
+        expectedBlockNumber: blockNumber,
+        expectedNounId: nounId,
+      });
```

Comment:
Couldn't test it with Sepolia, because `@dapp/core` doesn't support this test chain. I expect something will be off here, but it should be relatively easy to fix/update once we have mainnet contract we can use for test.